### DISCUSSION TITLE
a to link

### DISCUSSION THREAD

**Discussion by hajpepe** [vercel/next.js#37691] [Help]
```
  <div className={tw(`mr-5 flex items-baseline space-x-4`)}>
                {links.map((link: Link) => (
                  <a
                    key={link.label}
                    href={link.href}
                    className={tw(`text-gray-500 hover:text-gray-600 px-3 py-2 rounded-md font-medium`)}
                  >
                    {link.label}
                  </a>
                ))}
              </div>

```
 
how can i change `<a>` to `<Link>` ? for this code ?

**Comment by icyJoseph** [comment]
I removed the, in my opinion, unnecessary type annotation `link: Link`.

```tsx
import Link from 'next/link'; 

// later on in the JSX

<div className={tw(`mr-5 flex items-baseline space-x-4`)}>
  {links.map((link) => (
    <Link href={link.href} key={link.label}>
      <a
        className={tw(
          `text-gray-500 hover:text-gray-600 px-3 py-2 rounded-md font-medium`
        )}
      >
        {link.label}
      </a>
    </Link>
  ))}
</div>
```

**Comment by HellCatVN** [comment]
I want to improve above solution Link also have property that `passHref` which will send ref to a tag for better SEO

```js
import Link from 'next/link'; 

<div className={tw(`mr-5 flex items-baseline space-x-4`)}>
  {links.map((link) => (
    <Link href={link.href} key={link.label} passHref>
      <a
        className={tw(
          `text-gray-500 hover:text-gray-600 px-3 py-2 rounded-md font-medium`
        )}
      >
        {link.label}
      </a>
    </Link>
  ))}
</div>
```

**Reply by icyJoseph** [reply]
That's only necessary if you use a custom component, in which case that custom component has to also make use of `React.forwardRef`, because otherwise the href won't be passed. 

In OP's case,  the `href` is automatically passed to the `a` tag, so the `passHref` property is not needed.

There's some upcoming changes, not released yet, that'll make it so that you don't need the nested `a` tag, for all cases anymore. Today, if the children is a string, and you don't need CSS class-names, you could skip the a tag. 

In OP's case the `a` tag is necessary.

Please refer to docs, https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag.
